ui/StatusQ/src/modelutilsinternal.cpp

```diff
@@ -0,0 +1,56 @@
+#include "StatusQ/modelutilsinternal.h"
+
+#include <QAbstractItemModel>
+
+ModelUtilsInternal::ModelUtilsInternal(QObject* parent)
+    : QObject(parent)
+{
+}
+
+QStringList ModelUtilsInternal::roleNames(QAbstractItemModel *model) const
+{
+    if (model == nullptr)
+        return {};
+
+    QHash<int, QByteArray> roles = model->roleNames();
```

Comment:
DRY, `const auto ... = `

Reply:
Setting type explicitly here is not against DRY bc the type is not repeated. Imo it improves readability because you don't need to remember the return type of `roleNames` to understand the rest of the function body.

Reply:
Same as above, `auto` is always better; if it compiles, it's correct. Imagine you decide at some point to change the impl of the RHS to return `QMap`. With `auto` this would still work

Reply:
Ah right, then the return type has to be a QByteArrayList

Reply:
Yeah, but `QByteArrayList` is not qml expects so we have to stay with the current approach.